lua/nvim-navbuddy/actions.lua

```diff
@@ -278,22 +303,47 @@ function actions.comment(display)
 		return
 	end
 
+    local start_line = display.focus_node.scope["start"].line
+    local start_character = display.focus_node.scope["start"].character
+    local end_line = display.focus_node.scope["end"].line
+    local end_character = display.focus_node.scope["end"].character
+
 	display.state.leaving_window_for_action = true
 	vim.api.nvim_set_current_win(display.for_win)
 	vim.api.nvim_buf_set_mark(
 		display.for_buf,
 		"<",
-		display.focus_node.scope["start"].line,
-		display.focus_node.scope["start"].character,
-		{}
-	)
-	vim.api.nvim_buf_set_mark(
-		display.for_buf,
-		">",
-		display.focus_node.scope["end"].line,
-		display.focus_node.scope["end"].character,
+		start_line,
+		start_character,
 		{}
 	)
+    if end_character > 0 then
+        vim.api.nvim_buf_set_mark(
+            display.for_buf,
+            ">",
+            end_line,
+            end_character,
+            {}
+        )
+    else
+        if (end_line - start_line) > 0 then
+            local line_len = string.len(
+                vim.api.nvim_buf_get_lines(
+                    display.for_buf,
+                    end_line - 2,
+                    end_line - 1,
+                    false
+                )[1]
+            )
+            vim.api.nvim_buf_set_mark(
+                display.for_buf,
+                ">",
+                end_line - 1,
+                line_len - 1,
+                {}
+            )
+        end
+    end
```

Comment:
Can we avoid even this repeated code. Maybe instead of name_range_v_command and scope_v_command function we just have a function to handle the setting of cursor at end of provided scope? That could be more generally used in all actions.